client/src/components/ImageHandler.tsx

```diff
@@ -0,0 +1,98 @@
+import 'react-image-crop/dist/ReactCrop.css';
+
+import { useRef, useState } from 'react';
+import { twMerge } from 'tailwind-merge';
+
+import { ImageEditDialog } from './dialogs/ImageEditDialog';
+import { Button } from './ui/button';
+
+interface ImageHandlerProps {
+  callback?: (imageBlob: Blob, imageURL: string) => void;
+  removeSelectionCallback: () => void;
+  width: number;
+  height: number;
+  handlerId: string;
+  className?: string;
+}
+
+// callback will be run after the image is saved and is used to synchronize the component with its environment
+export function ImageHandler({
+  callback,
+  removeSelectionCallback,
+  width,
+  height,
+  className,
+  handlerId
+}: ImageHandlerProps) {
+  const [selectedImage, setSelectedImage] = useState<File>();
+  const [selectedImageURL, setSelectedImageURL] = useState<string>();
+  const filePickerRef = useRef<HTMLInputElement>(null); // unfortunately value for input type="file" component can't be set programatically, so have to use ref
```

Comment:
Can you please point the location where this ref is necessary? because I can't find it (probable skill issue)